go/pkg/reader/reader.go

```diff
@@ -148,8 +148,16 @@ type compressedSeeker struct {
 	fs   ReadSeeker
 	encd *zstd.Encoder
 	// This keeps the compressed data
-	buf    *syncedBuffer
-	closed bool
+	buf *syncedBuffer
+	err error
+}
+
+var encoderInit sync.Once
+var encoders sync.Pool
+
+func newEncoder() interface{} {
```

Comment:
I think this is leaky: zstd.NewWriter spawns goroutines that refer to some internal data, then when the zstd.Encoder is purged from the sync.Pool that internal data + goroutines leak.

Here is an example workaround: https://github.com/mostynb/zstdpool-syncpool

And some background: https://github.com/klauspost/compress/issues/264

Reply:
I missed not only the goroutines would leak but also the encoder itself since GC wouldn't ever catch it.

I imported your lib in - thanks!

Reply:
No problem- feedback welcome. It hasn't been thoroughly tested yet, but I figure it makes sense to share the effort in figuring out how to use this zstd api safely/efficiently (hence the tiny lib).

Reply:
Will certainly send PRs if I find any issues :)